tests/classes/tssubmission_test.php

```diff
@@ -1793,7 +1823,37 @@ public function test_similarity_overrides_are_false() {
         $tssubmission->setcm($cm->id);
 
         // Verify that viewer permissions are true as the config values are set to true.
-        $overrides = $tssubmission->create_similarity_overrides();
+        $overrides = $tssubmission->create_similarity_overrides(TURNITINSIM_ROLE_INSTRUCTOR);
+        $this->assertTrue($overrides['modes']['match_overview']);
+        $this->assertTrue($overrides['modes']['all_sources']);
+        $this->assertFalse($overrides['view_settings']['save_changes']);
+    }
+
+
+    /**
+     * Test that the similarity overrides are true when configured as such.
+     */
+    public function test_similarity_overrides_save_change_is_false_when_role_is_learner() {
+        $this->resetAfterTest();
+
+        set_config('turnitinviewersavechanges', 1, 'plagiarism_turnitinsim');
+
+        // Create assign module.
+        $record = new stdClass();
+        $record->course = $this->course;
+        $module = $this->getDataGenerator()->create_module('assign', $record);
+
+        // Get course module data.
+        $cm = get_coursemodule_from_instance('assign', $module->id);
+
+        // Create submission object.
+        $tssubmission = new plagiarism_turnitinsim_submission();
+        $tssubmission->setcm($cm->id);
+
+        // Verify that viewer permissions are true as the config values are set to true.
+        $overrides = $tssubmission->create_similarity_overrides(TURNITINSIM_ROLE_LEARNER);
+        $this->assertTrue($overrides['modes']['match_overview']);
```

Comment:
Please correct me if I am wrong but should this and the next one not be false for a learner?

Reply:
Yeah, I have discussed with Jake yesterday, these are fine, They are supposed to be true for every role.
It is similiar to CPF.